source/windows/iocp/socket.c

```diff
@@ -596,14 +597,16 @@ static int s_ipv4_stream_connection_success(struct aws_socket *socket) {
     struct iocp_socket *socket_impl = socket->impl;
     if (getsockopt(
             (SOCKET)socket->io_handle.data.handle, SOL_SOCKET, SO_ERROR, (char *)&connect_result, &result_length) < 0) {
+        int last_error = WSAGetLastError();
+        aws_win_log_message(AWS_LL_ERROR, AWS_LS_IO_SOCKET, "getsockopt()", last_error);
         AWS_LOGF_ERROR(
             AWS_LS_IO_SOCKET,
             "id=%p handle=%p: failed to determine connection error %d",
             (void *)socket,
             (void *)socket->io_handle.data.handle,
-            (int)WSAGetLastError());
+            (int)last_error);
         socket->state = ERRORED;
-        int error = s_determine_socket_error(WSAGetLastError());
+        int error = s_determine_socket_error(last_error);
```

Comment:
ok. this is my last comment. you're seeing the pattern. If the 2nd log statement is a repeat of the 1st, just cut it. If the 2nd log statement has new information (like this one reports id and handle) then log the aws-error, not just the %d of what you logged before